### ISSUE TITLE
FCA candidates showing when filtering for TBB in Partners field

### ISSUE THREAD

**Issue by samschlicht** [talentbeyondboundaries/tbbtalentv2#1292]
Hi <@U055YR1QT0A> <@U02NYRN57V5>,
I hope you're doing well. I was searching on TC and selected "Talent Beyond Boundaries" as the sole field for "Partners." However, the search results included candidates from FCA. Could you please look into this.  Thank you!

[Slack Message](https://refugeejobsmarket.slack.com/archives/C42S4AJ73/p1724141426018799?thread_ts=1724141426.018799&cid=C42S4AJ73)

**Comment by samschlicht** [comment]
The issue here is that direct DB edits of a candidate's managing partner are causing the ES index to go out of sync.

The candidate that Hanaa identified had TBB as partner in the ES index and FCA Uganda in the Postgres DB. 
That will be the case for quite a few candidates whose managing partner has been reassigned by direct DB edit, and then subsequently have not been interacted with in a way that would trigger the 'updateCandidateEs' step, thus updating the corresponding ES index entry.

There's nothing more to do on this particular instance: the ES reindexing for V2.2.2 will bring the DBs back into sync, and the implementation of #1215 will ensure that this doesn't happen again.

**Comment by samschlicht** [comment]
Note from Sadat:
We identified the issue: the user has "FCA" as their partner in Postgres but "TBB" in Elasticsearch. This mismatch likely occurred due to a direct edit in Postgres that wasn’t reflected in Elasticsearch. As a result, searches using Elasticsearch (like Hannah's) return results with "TBB" as the partner, but these are displayed with "FCA" from Postgres.
We propose creating a system admin API to support bulk partner updates for a given list, as Zeynep (or others) typically provide a list for such requests. Sam is adding a Sprint ticket for this. The next release will include an ES index update so Postgres and ES will be in sync, and the system admin API will help maintain that sync.
Separately, I'll deploy the caching time-to-live Hotfix to production, which we discussed this morning. While unrelated to Hannah’s issue, it'll provide a caching failsafe, which I think is worth doing.
